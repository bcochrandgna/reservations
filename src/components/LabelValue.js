// @flow
import React from 'react';
import { View, Text } from 'react-native';
import styles from '../styles/components/LabelValue';

type Props = {
  label: string,
  value: string,
};

const LabelValue = (props: Props) => {
  const { label, value } = props;

  return (
    <View style={styles.container}>
      <Text style={styles.labelText}>{`${label}:`}</Text>
      <Text style={styles.valueText}>{value}</Text>
    </View>
  );
};

export default LabelValue;
