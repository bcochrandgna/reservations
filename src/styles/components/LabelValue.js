import { StyleSheet } from 'react-native';
import { standardMargin, standardBold } from '..';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  labelText: {
    fontWeight: standardBold,
  },
  valueText: {
    marginLeft: standardMargin,
  },
});
