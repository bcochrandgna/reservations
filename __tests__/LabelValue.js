import React from 'react';
import { shallow } from 'enzyme';
import LabelValue from '../src/components/LabelValue';

describe('LabelValue', () => {
  it('renders correctly with null props', () => {
    const wrapper = shallow(<LabelValue />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with props', () => {
    const wrapper = shallow(<LabelValue label="Test Label" value="Test Value" />);
    expect(wrapper).toMatchSnapshot();
  });
});
