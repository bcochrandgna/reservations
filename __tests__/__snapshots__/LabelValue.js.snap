// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelValue renders correctly with null props 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      Object {
        "fontWeight": "700",
      }
    }
  >
    undefined:
  </Text>
  <Text
    style={
      Object {
        "marginLeft": 5,
      }
    }
  />
</View>
`;

exports[`LabelValue renders correctly with props 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      Object {
        "fontWeight": "700",
      }
    }
  >
    Test Label:
  </Text>
  <Text
    style={
      Object {
        "marginLeft": 5,
      }
    }
  >
    Test Value
  </Text>
</View>
`;
